scripts/components/Explore.js

```diff
@@ -1,39 +1,82 @@
 'use strict';
 
-import React from 'react';
-import LinkedStateMixin from 'react/lib/LinkedStateMixin';
-import { Navigation } from 'react-router';
+import React, { PropTypes } from 'react';
+import shouldComponentUpdatePure from '../utils/shouldComponentUpdatePure';
 
-const Explore = React.createClass({
-  mixins: [LinkedStateMixin, Navigation],
+const DEFAULT_LOGIN = 'gaearon';
+const GITHUB_REPO = 'https://github.com/gaearon/flux-react-router-example';
 
-  getInitialState() {
-    return {
-      loginOrRepo: 'gaearon'
-    };
-  },
+function parseFullName(params) {
+  if (!params.login) {
+    return DEFAULT_LOGIN;
+  }
+  return params.login + (params.name ? '/' + params.name : '');
+}
+
+export default class Explore extends React.Component {
+
+  shouldComponentUpdate = shouldComponentUpdatePure
+
+  static propTypes = {
+    params: PropTypes.shape({
+      login: PropTypes.string,
+      name: PropTypes.string
+    })
+  }
+
+  static contextTypes = {
+    router: PropTypes.func.isRequired
+  }
 
+  constructor() {
+    super();
+
+    this.handleKeyUp = this.handleKeyUp.bind(this);
+    this.handleGoClick = this.handleGoClick.bind(this);
+  }
+
+  // FIXME: update input when URL changes.
+  //
+  // What we want is the `input` field to reflect
+  // the current URL and still be editable by the user.
+  //
+  // Since we want the user to edit the field, we have to
+  // use an `uncontrolled` component (and provide a
+  // `defaultValue` to initialize the field).
+  //
+  // Problem here is that we can't programmatically update
+  // the field with other values. To do that, we would have
+  // to use a `controlled` component with a `value`,
+  // but this will prevent the user to change the field.
+  //
+  // To enable user input in this case, we have to provide
+  // an `onChange` event, which is triggered whenever the
+  // user types something (and we can't check the key event here).
+  //
+  // Any idea?
```

Comment:
@gaearon I've described here the problem with `input`. Do you have any idea how we can solve this?